src/api.ts

```diff
@@ -325,6 +325,63 @@ type CreateAttachmentParams = {
   metadata?: Maybe<Record<string, any>>;
 };
 
+export class SharedCache {
+  private static instance: SharedCache | null = null;
+  private cache: Map<string, any>;
+
+  private constructor() {
+    this.cache = new Map();
+  }
+
+  static getInstance(): SharedCache {
+    if (!SharedCache.instance) {
+      SharedCache.instance = new SharedCache();
+    }
+    return SharedCache.instance;
+  }
+
+  public getPromptCacheKey(
+    id?: string,
+    name?: string,
+    version?: number
+  ): string {
+    if (id) {
+      return id;
+    } else if (name && (version || version === 0)) {
+      return `${name}:${version}`;
+    } else if (name) {
+      return name;
+    }
+    throw new Error('Either id or name must be provided');
+  }
+
+  public getPrompt(key: string): Prompt {
```

Comment:
This is no longer needed

Reply:
Is doing `cache.get() as Prompt` better?